server/src/main/java/com/tencent/rss/server/storage/HdfsStorageManager.java

```diff
@@ -87,6 +87,14 @@ public Checker getStorageChecker() {
   public void registerRemoteStorage(String appId, String remoteStorage) {
     if (!pathToStorages.containsKey(remoteStorage)) {
       pathToStorages.putIfAbsent(remoteStorage, new HdfsStorage(remoteStorage, hadoopConf));
+      // registerRemoteStorage may be called in different threads,
+      // make sure metrics won't be created duplicated
+      // there shouldn't have performance issue because
+      // it will be called only few times according to the number of remote storage
+      synchronized (this) {
```

Comment:
maybe addDynamicCounterForRemoteStorage should be synchronized instead of HdfsStorageManager.

Reply:
ok